src/main/java/org/skriptlang/reflect/syntax/condition/elements/StructCustomCondition.java

```diff
@@ -0,0 +1,186 @@
+import org.skriptlang.reflect.syntax.condition.ConditionCheckEvent;
+import org.skriptlang.reflect.syntax.condition.ConditionSyntaxInfo;
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.script.Script;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class StructCustomCondition extends CustomSyntaxStructure<ConditionSyntaxInfo> {
+
+  public static boolean customConditionsUsed = false;
+
+  static {
+    String[] syntax = {
+        "[(1¦local)] condition <.+>",
+        "[(1¦local)] condition",
+        "[(1¦local)] %*classinfos% property condition <.+>"
+    };
+    Skript.registerStructure(StructCustomCondition.class, customSyntaxValidator()
+        .addSection("check", true)
+        .build(), syntax);
+  }
+
+  private static final DataTracker<ConditionSyntaxInfo> dataTracker = new DataTracker<>();
+
+  static final Map<ConditionSyntaxInfo, Trigger> conditionHandlers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, Trigger> parserHandlers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, List<Supplier<Boolean>>> usableSuppliers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, Boolean> parseSectionLoaded = new HashMap<>();
+
+  static {
+    Skript.registerCondition(CustomCondition.class);
+    Optional<SyntaxElementInfo<? extends Condition>> info = Skript.getConditions().stream()
+        .filter(i -> i.c == CustomCondition.class)
+        .findFirst();
+    info.ifPresent(dataTracker::setInfo);
+
+    dataTracker.addManaged(conditionHandlers);
+    dataTracker.addManaged(parserHandlers);
+    dataTracker.addManaged(usableSuppliers);
+    dataTracker.addManaged(parseSectionLoaded);
+  }
+
+  private SectionNode parseNode, checkNode;
+  private Runnable register;
+
+  @Override
+  public DataTracker<ConditionSyntaxInfo> getDataTracker() {
+    return dataTracker;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public boolean init(Literal<?>[] args, int matchedPattern, SkriptParser.ParseResult parseResult,
+                      EntryContainer entryContainer) {
+    customConditionsUsed = true;
+
+    SectionNode patterns = entryContainer.getOptional("patterns", SectionNode.class, false);
+    Script script = (parseResult.mark & 1) == 1 ? SkriptUtil.getCurrentScript() : null;
+
+    switch (matchedPattern) {
+      case 0:
+        String pattern = parseResult.regexes.get(0).group();
+        register = () -> register(ConditionSyntaxInfo.create(script, pattern, 1, false, false));
+        break;
+      case 1:
+        if (patterns == null) {
+          Skript.error("Custom conditions without inline patterns must have a patterns section.");
+          return false;
+        }
+
+        register = () -> {
+          int i = 1;
+          for (Node subNode : patterns) {
```

Comment:
This does not check for there being no subnodes, while other custom syntax implementations do. Maybe you could add a utility method to CustomSyntaxStructure?